.circleci/config.yml

```diff
@@ -528,13 +528,16 @@ workflows:
             branches:
               only: main
     jobs:
-      - build_docker_image
+      - build_docker_image:
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8", "3.9"]
       - build_kedro:
           matrix:
             parameters:
-              python_version: ["3.6", "3.7", "3.8"]
+              python_version: ["3.6", "3.7", "3.8", "3.9"]
           requires:
-            - build_docker_image
+            - build_docker_image-<<matrix.python_version>>
```

Comment:
This means that `build_kedro-3.6` only depends on `build_docker_image-3.6`, etc.